source/extensions/filters/common/expr/custom_cel/extended_request/custom_cel_attributes.cc

```diff
@@ -0,0 +1,57 @@
+#include "source/extensions/filters/common/expr/custom_cel/extended_request/custom_cel_attributes.h"
+
+#include "source/common/http/utility.h"
+#include "source/extensions/filters/common/expr/context.h"
+
+#include "eval/public/cel_value.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace CustomCel {
+namespace ExtendedRequest {
+
+using google::api::expr::runtime::CreateErrorValue;
+using Http::Utility::parseAndDecodeQueryString;
+using Http::Utility::QueryParams;
+
+// ExtendedRequestWrapper extends RequestWrapper
+// If ExtendedRequestWrapper[key] is not found, then the base RequestWrapper[key] is returned.
+absl::optional<CelValue> ExtendedRequestWrapper::operator[](CelValue key) const {
```

Comment:
@kyessenov can you review the CEL specifics in this PR? No need to worry about the general extension parts.